test-requirements.txt

```diff
@@ -1 +1,17 @@
-PyMySQL
+coverage!=4.4,>=4.0 # Apache-2.0
+fixtures>=3.0.0 # Apache-2.0/BSD
+python-subunit>=1.0.0 # Apache-2.0/BSD
+testtools>=2.2.0 # MIT
+testresources>=2.0.0 # Apache-2.0/BSD
+testscenarios>=0.4 # Apache-2.0/BSD
+WebTest>=2.0.27 # MIT
+oslotest>=3.2.0 # Apache-2.0
+stestr>=1.0.0 # Apache-2.0
+ddt>=1.0.1 # MIT
+# Needed to run DB commands in virtualenvs
+PyMySQL>=0.7.6 # MIT License
+pytest
+pytest-mock
+pytest-flask
+# (rlin): new released 8.4.0 still got async bug
+tenacity<=8.3.0
```

Comment:
I see a bunch of these not actually references in the project, I'd like for us to trim these down to the absolute necessary.